### ISSUE TITLE
Decide roles and aria attributes for the cookie banner

### ISSUE THREAD

**Issue by hannalaakso** [alphagov/govuk-frontend#2102]
## What
Work out what attributes and roles we want to set on the component. 

Where possible and appropriate, it should be consistent with the attributes on our existing components.

There are a few things we need to consider:

- Do we agree that `region` is the most appropriate role to use? 
- Do we set an `aria-label`? What should the default text be?
- Do we set `role="alert"` on the `<p>` inside 'You’ve accepted all cookies' banner, similar to what GOV.UK does? We do need to make sure that the confirmation is read out by AT.
- Do we need to `tabindex=-1` on the 'You’ve accepted all cookies' banner, similar to what GOV.UK does?
- Are there any other accessibility roles or attributes we should consider using?

Once we have agreed the attributes we plan to test the component with different assistive technologies and to take it to the GDS Accessibility clinic for additional reassurance.

## Why
By agreeing the roles and aria attributes we'll have a version we can test to see how it behaves with different assistive technologies.

## Who needs to know about this
Developers

## Done when

- [x] We've watched Leonie's talk on [Screen readers and cookie consents](https://ldnwebperf.org/sessions/screen-readers-and-cookie-consents/)
- [x] We've decided on the roles and attributes for the cookie banner
- [x] We've documented our decision making in a draft form which we can present at the accessibility clinic if necessary

**Comment by vanitabarrett** [comment]
## Role attribute
Looking through the [MDN docs for aria roles](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles), `region` seems to be the most appropriate. Definition of the region role:

> A perceivable section containing content that is relevant to a specific author-specified purpose and sufficiently important the users will likely want to be able to navigate to the section easily and have it listed in the summary of the page

The only other option which stood out was `banner`, but this is more about the common banner/header at the top of a site which contains the logo, navigation, site name etc. Some sites have combined their cookie banner in with their existing header, for example: [BBC](https://github.com/bbc/simorgh/issues/1707), but this would probably be difficult for us to implement given our header component has its own `role=banner`.

The `role=region` should probably apply to the parent wrapper that goes around all banner (question + confirmation) HTML, so that a user can use landmarks to navigate back to it even after clicking accept/reject. Example:

```
<div role='region'>
    <div class='govuk-cookie-banner__question-banner'>...</div>
    <div class='govuk-cookie-banner__confirmation-banner'>...</div>
</div>
```

## Label
Aria states: 

> Authors MUST give each element with role region a brief label that describes the purpose of the content in the region. Authors SHOULD reference a visible label with aria-labelledby if a visible label is present. Authors SHOULD include the label inside of a heading whenever possible.

For the question banner, there is a visible label: the banner heading. However, this disappears once a user has pressed accept/reject, so I'm not sure we can use that. **Perhaps the heading text could be passed into `aria-label` unless a different label is specified via an aria-label macro option?**

```
<div role='region' aria-label='Cookies on GOV.UK'>
    <div class='govuk-cookie-banner__question-banner'>
        <h1>Cookies on GOV.UK</h1>
    </div>
    <div class='govuk-cookie-banner__confirmation-banner'>...</div>
</div>
```

**Comment by vanitabarrett** [comment]
Tested the above proposed attributes with [the prototype](https://github.com/alphagov/govuk-frontend/pull/2081/commits/1ba0dd1b8596b735e6bf024364712ffa216129b6) in various assistive tech and it largely worked as expected.

Client Side:
- Voiceover: Focuses and reads out confirmation banner
- NVDA + Firefox: Focuses and reads out confirmation banner
- NVDA + Chrome: Focuses and reads out confirmation banner
- Narrator + Edge: Focuses and reads out confirmation banner
- JAWS: TBC
- Android TalkBack: Focuses and reads out confirmation banner

Server Side:
- Voiceover: Read the message out twice
- NVDA + Firefox: Read the cookie banner out, but perhaps only because it was at the top of the page Then reads out rest of page.
- NVDA + Chrome:  Reads out ‘alert’ and the confirmation message. Then reads rest of page.
- Narrator + Edge: Focuses and reads out confirmation banner
- JAWS: TBC
- Android TalkBack: Focuses and reads out confirmation banner

Note: for some reason, JAWS in AssistivLabs wasn't working for me so I was unable to test.

**Comment by vanitabarrett** [comment]
TLDR: I think we want `role=region` and `aria-label` on the cookie banner. And for the confirmation banner, we probably want to follow what we did for notification banner, so `role=alert` `tabindex='-1'` and some additional JS to set focus.

Proposed default aria-label text for now is "Cookie banner"
